cpp/include/cudf/io/json.hpp

```diff
@@ -146,14 +155,35 @@ class json_reader_options {
    *
    * @param types Vector dtypes in string format.
    */
-  void dtypes(std::vector<std::string> types) { _dtypes = std::move(types); }
+  [[deprecated(
+    "The string-based interface will be deprecated."
+    "Use dtypes(std::vector<data_type>) or "
+    "dtypes(std::map<std::string, data_type>) instead.")]] void
+  set_dtypes(std::vector<std::string> types)
+  {
+    _dtypes = std::move(types);
+  }
+
+  /**
+   * @brief Set data types for columns to be read.
+   *
+   * @param types Vector of dtypes
+   */
+
+  void set_dtypes(std::vector<data_type> types) { _dtypes = std::move(types); }
+  /**
```

Comment:
```suggestion

    /**
```
nit: add empty line here?